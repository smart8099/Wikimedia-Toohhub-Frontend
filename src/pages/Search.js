import React, { useState } from 'react';
import SampleToolsData from './SampleToolsData';

const Search = () => {
    const [searchValue, setSearchValue] = useState('');
    const [toolsData] = useState(SampleToolsData);
    const [allTools, setAllTools] = useState([]);
    const handleChange = (e) => {
        try {
            setSearchValue(e.target.value);
        } catch (error) {
            console.log(error); //eslint-disable-line
        }
    };

    const handleSearch = (e) => {
        try {
            e.preventDefault();
            if (!(searchValue == '')) {
                const filteredTools = toolsData.filter(
                    (tool) => tool.name.toLowerCase().includes(
                        searchValue.toLowerCase(),
                    ),
                );
                setAllTools(filteredTools);        

            }
            else{
                setAllTools([])
            }
            
            
            console.log(filteredTools); //eslint-disable-line
        } catch (error) {
            console.log(error); //eslint-disable-line
        }
    };

    const handleReset = () => {
        try {
            setAllTools([]);
        } catch (error) {
            console.log(error); //eslint-disable-line
        }
    };

    const renderSearchResults = allTools.map((tool) => (
        <div className="search--results--container" key={tool.id}>
            <div className="search--results--title">
                <h3>{tool.name}</h3>
            </div>
            <div className="search--results--description">
                <p>{tool.description}</p>
            </div>
        </div>
    ));

    return (
        <>
            <div className="search-field">


                <div className="input-group">
                    <input
                        className="form-control"
                        type="text"
                        placeholder="Search for a tool"
                        onChange={handleChange}

                    />
                    <div className="button-container">
                        <button className="" onClick={handleSearch} type="button">
                            Search
                        </button>
                        <button className="" onClick={handleReset} type="button">
                            Reset
                        </button>
                    </div>
                </div>
               {renderSearchResults} 
            </div>
        </>
    );
};

export default Search;