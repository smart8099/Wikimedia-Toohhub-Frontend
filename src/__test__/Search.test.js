import React from "react";
import renderer from 'react-test-renderer';
import Search from '../pages/Search'

describe('Search', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<Search />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });