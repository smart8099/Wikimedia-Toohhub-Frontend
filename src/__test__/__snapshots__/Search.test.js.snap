// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search-field"
>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search for a tool"
      type="text"
    />
    <div
      className="button-container"
    >
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
